src/components/AddAssetsBottomSheet.tsx

```diff
@@ -50,23 +55,52 @@ export default function AddAssetsBottomSheet({
       iconComponent: QuickActionsSwap,
       title: t('addFundsActions.swap'),
     },
+    [TokenActionName.SwapAndDeposit]: {
+      iconComponent: QuickActionsSwapAndDeposit,
+      title: t('addFundsActions.swapAndDeposit'),
+    },
+    [TokenActionName.CrossChainSwap]: {
+      iconComponent: QuickActionsSwap,
+      title: t('addFundsActions.crossChainSwap'),
+    },
   }
 
   const addAssetsActions = actions.map((action) => ({
     ...action,
     ...actionExtraProps[action.name],
   }))
 
+  const beforeDescriptionAction = addAssetsActions[0]
+  const afterDescriptionActions = addAssetsActions.slice(showDescriptionAfterFirstAction ? 1 : 0)
```

Comment:
This is very confusing. Can you instead just treat the description as part of an individual action and leave it undefined for all but the first or second action?